src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -2425,7 +2425,9 @@ void ShenandoahHeap::set_concurrent_old_mark_in_progress(bool in_progress) {
   // has_forwarded_objects() iff UPDATEREFS or EVACUATION
   bool has_forwarded = has_forwarded_objects()? 1: 0;
   bool updating_or_evacuating = _gc_state.is_set(UPDATEREFS | EVACUATION)? 1: 0;
-  assert (has_forwarded == updating_or_evacuating, "Has forwarded objects iff updating or evacuating");
+  bool evacuating = gc_state.is_set(EVACUATION)? 1: 0;
```

Comment:
suggestion: 
`
bool evacuating = gc_state.is_set(EVACUATION);
`
pre-exist:
`bool has_forwarded = has_forwarded_objects()? 1: 0;` to `bool has_forwarded = has_forwarded_objects();`

Reply:
Thanks.  I'll fix those, and my misspelling of gc_state.